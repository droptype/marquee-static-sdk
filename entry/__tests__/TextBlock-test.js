const React = require('react')
const TextBlock = require('../TextBlock')
const renderer = require('react-test-renderer')

const SAMPLE_BLOCK = require('./SAMPLE_ENTRY.json').content.find( block => 'text' === block.type)

describe('TextBlock', () => {
    it('renders', () => {
        const component = renderer.create(
            React.createElement(TextBlock, { block: SAMPLE_BLOCK })
        )
        expect(
            component.toJSON()
        ).toMatchSnapshot()
    })
    it('renders plain', () => {
        const component = renderer.create(
            React.createElement(TextBlock, { block: SAMPLE_BLOCK, plain: true })
        )
        expect(
            component.toJSON()
        ).toMatchSnapshot()
    })
})
