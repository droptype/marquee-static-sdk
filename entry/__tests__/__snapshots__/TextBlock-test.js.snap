// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBlock renders 1`] = `
<p
  className={
    Classes {
      "_groups": Object {
        "align": "left",
        "role": "paragraph",
      },
      "_initial": Array [
        "Block",
        "TextBlock",
      ],
      "_switches": Object {},
    }
  }
  dangerouslySetInnerHTML={
    Object {
      "__html": "Nullam quis <strong class=\\"Annotation -strong\\">risus</strong> eget urna mollis ornare vel eu leo. Donec sed odio dui. Sed posuere <em class=\\"Annotation -emphasis\\">consectetur</em> est at lobortis. Cum sociis natoque <strong class=\\"Annotation -strong\\">penatibus <em class=\\"Annotation -emphasis\\">et magnis</em></strong><em class=\\"Annotation -emphasis\\"> dis</em> parturient montes, nascetur ridiculus mus.",
    }
  }
  id="text_7d112f487ec748c99aa022c7298c6965"
/>
`;

exports[`TextBlock renders plain 1`] = `
<p
  dangerouslySetInnerHTML={
    Object {
      "__html": "Nullam quis <strong>risus</strong> eget urna mollis ornare vel eu leo. Donec sed odio dui. Sed posuere <em>consectetur</em> est at lobortis. Cum sociis natoque <strong>penatibus <em>et magnis</em></strong><em> dis</em> parturient montes, nascetur ridiculus mus.",
    }
  }
/>
`;
